netsocadmin/routes/signup.py

```diff
@@ -157,6 +159,116 @@ def dispatch_request(self) -> str:
+            return flask.render_template(
+                "index.html",
+                page="login",
+                error_message="An error occured. Please try again or contact us",
+            )
+
+        template = "message.html"
+        caption = "Success!"
+        message = f"An email has been sent with your new password. Please change your password as soon as you log in."
+        # message = f"<p>{reset_resp}</p>" + message;
+        return flask.render_template(template, caption=caption, message=message)
+
+    def dispatch_request(self) -> str:
+        # Make sure they haven't forged the URL
+        user = flask.request.args.get("u")
+        email = flask.request.args.get("e")
+        with sentry_sdk.configure_scope() as scope:
+            scope.user = {
+                "email": email,
+            }
+
+        token = flask.request.args.get("t")
+        return self.render(user, email, token, not register_tools.good_token(email, token))
+
+class Forgot(View):
+    """
+    Route: /forgot
+        Users will be lead to this route when they submit an email for a reminder of their
+        username and, optionally, resetting their password.
+        It then checks that form data to make sure it's a valid UCC email.
+        Sends an email with a link to validate the email holder is who is registering.
+    """
+    # Logger instance
+    logger = logging.getLogger("netsocadmin.forgot")
+    # Specify which method(s) are allowed to be used to access the route
+    methods = ["POST"]
+
+    def dispatch_request(self) -> str:
+        # make sure is ucc email
+        email = flask.request.form['email']
+
+        with sentry_sdk.configure_scope() as scope:
+            scope.user = {
+                "email": email,
+            }
+
+        if not re.match(r"[0-9]{8,11}@umail\.ucc\.ie", email) and not re.match(r"[a-zA-Z.0-9]+@uccsocieties.ie", email):
+            self.logger.info(f"email {email} is not a valid UCC email")
+            return flask.render_template(
+                "index.html",
+                page="login",
+                error_message="Must be a UCC Umail or Society email address")
+
+        # make sure email has not already been used to make an account
+        if email in config.EMAIL_WHITELIST or not register_tools.has_account(email):
+            self.logger.info(f"account doesn't exist with email {email}")
+            return flask.render_template(
+                "message.html",
+                caption="Sorry!",
+                message=(
+                    f"There isn't an existing account with email '{email}'. "
+                    "Please contact us if you think this is an error."
+                ),
+            )
+
+        # send confirmation link to ensure they own the email account
+        out_email = (
+            "admin.netsoc.co" if not config.FLASK_CONFIG["debug"]
+            else f"{config.FLASK_CONFIG['host']}:{config.FLASK_CONFIG['port']}"
+        )
```

Comment:
This is probably redundant. Its only used in `send_forgot_email`, where its ignored if config.FLASK_CONFIG['debug'] is True

Reply:
So I implemented that way because that's how it's done for "/sendconfirmation" (hence the "Account Registration" points which I'll fix). I'd agree that it's mostly redundant, but it does actually serve some purpose when developing in dev-env. Since we can't send emails in dev-env, the easiest way to test the email's body is the return the string and render it as the "message_html" message. This out_email code allows us to test the email body with working links without worrying about substituting "0.0.0.0" for "admin.netsoc.co". I think it's worth keeping for ease of debugging.